### ISSUE TITLE
"azurerm_cdn_endpoint_custom_domain" - Akamai - cannot enable HTTPS: "Cannot override minimum TLS version for Akamai since already has a minimum of 1.2"

### ISSUE THREAD

**Issue by alexs77** [hashicorp/terraform-provider-azurerm#15704]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v1.1.7

- hashicorp/azurerm v2.98.0

### Affected Resource(s)

* `azurerm_cdn_endpoint_custom_domain`

### Terraform Configuration Files

```hcl
resource "azurerm_cdn_endpoint_custom_domain" "app_services" {
  name            = "example"
  cdn_endpoint_id = var.endpoint_id
  host_name       = "cdn.example.net"

  cdn_managed_https {
    certificate_type = "Shared"
    protocol_type    = "ServerNameIndication"
  }
}
```

### Debug Output

`terraform plan`:

```text
  # module.cdn[0].module.cd["urihandler"].azurerm_cdn_endpoint_custom_domain.app_services[0] will be updated in-place
  ~ resource "azurerm_cdn_endpoint_custom_domain" "app_services" {
        id              = "/subscriptions/.../customDomains/assets"
        name            = "example"
        # (2 unchanged attributes hidden)

      + cdn_managed_https {
          + certificate_type = "Shared"
          + protocol_type    = "ServerNameIndication"
          + tls_version      = "TLS12"
        }
    }
```

### Expected Behaviour

The resource should be created.

### Actual Behaviour

It fails. Error message:

```text
Error: enable CDN Managed HTTPS on "Custom Domain: (Name \"assets\" / Endpoint Name \"prep\" / Profile Name \"cdn-prep\" / Resource Group \"PreProduction\")": sending enable request: cdn.CustomDomainsClient#EnableCustomHTTPS: Failure sending request: StatusCode=400 -- Original Error: Code="BadRequest" Message="Cannot override minimum TLS version for Akamai since already has a minimum of 1.2"

  with module.cdn[0].module.cd["prep"].azurerm_cdn_endpoint_custom_domain.app_services[0],
  on modules/cdn/modules/cd/main.tf line 21, in resource "azurerm_cdn_endpoint_custom_domain" "app_services":
  21: resource "azurerm_cdn_endpoint_custom_domain" "app_services" {

##[error]Error: The process '/opt/hostedtoolcache/terraform/1.1.7/x64/terraform' failed with exit code 1
##[section]Finishing: Terraform: validate & apply
```

Message text sent from Azure: **Cannot override minimum TLS version for Akamai since already has a minimum of 1.2**

### Important Factoids

Note that `terraform plan` showed this:

```
…
          + tls_version      = "TLS12"
…
```

Terraform tried to set "`tls_version = "TLS12"`".

Seems that this cannot be done.

**Comment by magodo** [comment]
The error message seems confusing, while one thing can worth noting is that the [`Akamai` sku doesn't support HTTPS](https://docs.microsoft.com/en-us/azure/cdn/cdn-custom-ssl?tabs=option-2-enable-https-with-your-own-certificate#tlsssl-certificates) for now.

**Comment by alexs77** [comment]
Okay?
Well, when manually setting up a CDN on portal, HTTPS does work. Or do I misunderstand you?

![image](https://user-images.githubusercontent.com/781361/156981052-88b83d19-cc6e-49a2-867f-0fb4158128e9.png)
![image](https://user-images.githubusercontent.com/781361/156981188-6d515941-ee1f-4f6d-ac55-04161e469e18.png)

```sh
$ curl -v https://$cdn_domain
…
* Server certificate:
*  subject: CN=cert00070-azurecdn.akamaized.net
*  start date: Mar  7 03:20:48 2022 GMT
*  expire date: Jun  5 03:20:47 2022 GMT
*  subjectAltName: host "…" matched cert's "…"
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify ok.
…
* old SSL session ID is stale, removing
* Connection state changed (MAX_CONCURRENT_STREAMS == 100)!
< HTTP/2 404 
< server: Microsoft-IIS/10.0
…
```

**Comment by magodo** [comment]
@alexs77 Thanks for the update! I'll check with the CDN service team.

**Comment by magodo** [comment]
@alexs77 This should be fixed by #15756, after which is merged, you can specify the `tls_version = "None"` in the `cdn_managed_https` block.

**Comment by magodo** [comment]
Closing this issue as is fixed in #16750
